File: api/fetch-preview.js
export default function handler(req, res) {
  const url = req.query.url;
  if (!url) {
    return res.status(400).json({ error: "No URL provided" });
  }

  try {
    const match = url.match(/(?:v=|youtu\.be\/)([a-zA-Z0-9_-]+)/);
    if (!match) {
      return res.status(400).json({ error: "Invalid URL" });
    }
    const video_id = match[1];
    return res.json({ video_id });
  } catch (err) {
    return res.status(500).json({ error: "Server error: " + err.message });
  }
}

File: api/fetch-preview.py
from flask import Flask, request, jsonify
import yt_dlp

app = Flask(__name__)

@app.route("/", methods=["GET"])
def fetch_preview():
    url = request.args.get("url")
    if not url:
        return jsonify({"error": "No URL provided"}), 400

    try:
        with yt_dlp.YoutubeDL({"quiet": True}) as ydl:
            info = ydl.extract_info(url, download=False)
            video_id = info.get("id")
        return jsonify({"video_id": video_id})
    except Exception as e:
        return jsonify({"error": f"Server error: {str(e)}"}), 500
